appinventor/components/src/com/google/appinventor/components/runtime/Texting.java

```diff
@@ -616,9 +620,32 @@ public void ReceivingEnabled(@Options(ReceivingState.class) int enabled) {
           ErrorMessages.ERROR_BAD_VALUE_FOR_TEXT_RECEIVING, enabled);
       return;
     }
+    if (state == ReceivingState.Always) {
+      checkForNotificationPermission("ReceivingEnabled");
```

Comment:
If the permission needs to be asked for, it seems like we will need to return here before proceeding to the call to `ReceivingEnabledAbstract` so that we only set it once the permission has been granted. There are other components like Camera that follow this pattern.

Reply:
Reverted

Reply:
Fixed